pkg/lib/cockpit-components-table.jsx

```diff
@@ -157,10 +159,7 @@ export class ListingTable extends React.Component {
         if (this.props['aria-label'])
             props['aria-label'] = this.props['aria-label'];
 
-        const tableBodyProps = {};
-        // We need the following because of: https://github.com/patternfly/patternfly-react/issues/3090
-        if (props.cells.some(cell => typeof (cell.title) != 'string' || cell.title.toLowerCase() == 'id' || !cell.title))
-            tableBodyProps.rowKey = ({ rowData, rowIndex }) => rowIndex;
+        const tableBodyProps = { rowKey: ({ rowData, rowIndex }) => (rowData.props && rowData.props.key) ? rowData.props.key : rowIndex };
```

Comment:
I'd rather not put a fallback here. `rowIndex` is wrong, and I'd rather have a React warning about a missing key, so that the issue is not silently covered.